### PR TITLE
ENH check tear sheet inputs

### LINKED ISSUE #211: plot_holding function calcuate the holding with some mistake
the mistake happens in line 345 in the plotting .py file:
```Python
    df_holdings = positions.apply(lambda x: np.sum(x != 0), axis='columns')
```
since ```np.NaN != 0```here
```
 np.sum(x != 0)
```
includes the case when position is NaN, which shoud be ignored when counting holdings. 
also I believe compare a float with zero is not a good practice.

### PR DESCRIPTION
**Pull request by eigenfoo**
Closes #211 

New dependency: `engarde>= 0.4.0` 

I made a new module with a bunch of functions to apply basic sanity checks to `positions`, `returns` and `transactions`. There's a master `check_inputs` function, which will run the appropriate checks on whatever is passed in. At the start of every `create_foo_tear_sheet` function, we just need to call `check_inputs`, and it will raise a special `TearSheetInputError` exception if anything's wrong.

Completely untested and not yet ready for review: I just wanted to kick off a Travis test to see if I broke anything. I'll play around with the example notebooks to see what this looks like when I get the chance later tonight/tomorrow.

*This PR is a bit confusing because it also has my commits from #427. This should stop being a problem once that PR is merged, though.*

### CODE REVIEW

**Comment by twiecki** [comment]
Wow, this is incredible. Will review soon!

File: .bumpversion.cfg [hunk 1]
```diff
@@ -1,7 +1,7 @@
 [bumpversion]
 commit = True
 tag = True
-current_version = 0.1
+current_version = 0.8.0
```
Comment (twiecki) [inline .bumpversion.cfg]: We don't use this file anymore, can be deleted.

File: pyfolio/check_inputs.py [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+    - Positions must be floats
+    """
+    try:
+        ed.none_missing(positions)
+    except AssertionError as e:
+        raise TearSheetInputError('''positions has NaNs. engarde raises the
+                                  following: ''' + e)
+
+    if positions.index.tz is None:
+        raise TearSheetInputError('positions index is not timezone-localized')
```
Comment (twiecki) [inline pyfolio/check_inputs.py]: would be even better if we say how to fix it.

**Comment by twiecki** [comment]
Seems like we use `engarde` in only a few places, do we really need it?

File: .bumpversion.cfg (hunk 1 above)
Comment (eigenfoo) [reply .bumpversion.cfg]: Will delete on the 0.8.0 PR.

**Comment by eigenfoo** [comment]
I was actually coming around to that conclusion myself: I'll take a look at what `engarde` actually does and see if we can write our own checks.

**Comment by eigenfoo** [comment]
It's been almost a year. Closing this PR for now, maybe we'll revisit it in the future.

**Comment by twiecki** [comment]
Actually I really liked this effort. Nevermind engarde. But we should automatically tz-convert and tell the user.

**Pull request closed without merging**